go/vt/sqlannotation/sqlannotation.go

```diff
@@ -67,51 +68,54 @@ func IsDML(sql string) bool {
 		strings.EqualFold(word, "delete")
 }
 
-// ExtractKeySpaceID parses the annotation of the given statement and tries
+// ExtractKeyspaceIDS parses the annotation of the given statement and tries
 // to extract the keyspace id.
 // If a keyspace-id comment exists 'keyspaceID' is set to the parsed keyspace id
 // and err is set to nil; otherwise, if a filtered-replication-unfriendly comment exists
 // or some other parsing error occured, keyspaceID is set to nil and err is set to a non-nil
 // error value.
-func ExtractKeySpaceID(sql string) (keyspaceID []byte, err error) {
-	keyspaceIDString, hasKeySpaceID := extractStringBetween(sql, "/* vtgate:: keyspace_id:", " ")
+func ExtractKeyspaceIDS(sql string) (keyspaceIDs [][]byte, err error) {
+	_, comments := sqlparser.SplitTrailingComments(sql)
+	keyspaceIDString, hasKeyspaceID := extractStringBetween(comments, "/* vtgate:: keyspace_id:", " ")
 	hasUnfriendlyAnnotation := (strings.Index(sql, filteredReplicationUnfriendlyAnnotation) != -1)
-	err = nil
-	if hasKeySpaceID {
+	if hasKeyspaceID {
+		ksidStr := strings.Split(keyspaceIDString, ",")
+		keyspaceIDs = make([][]byte, len(ksidStr))
 		if hasUnfriendlyAnnotation {
-			keyspaceID = nil
+			keyspaceIDs = nil
```

Comment:
I would just:
return nil, &ExtractKeySpaceIDError{
...
}
here, that avoids a level of indentation.

then the two lines ksidStr := ... and keyspaceIDs = ... can be moved after the check.